### DISCUSSION TITLE
Row selection with controlled pagination

### DISCUSSION THREAD

**Discussion by ignlopezsanchez** [TanStack/table#2661] [General] [answered]
Hi, I have a table with controlled pagination from server and row selection.
If a try to set autoResetSelectedRows: true as table option, when I select rows in a page => request a new page => the rows in the same position in the second page will be selected. I know this is the expected behavior becasue with controlled pagination, new pages start with reset index.

But, I am trying to implement the behaviour I want and don't know what is the best option. I would expect to have something like selectedRowIds and selectedFlatRows, as I am using them in other parts of my code.
Thanks!

**Comment by nicklas-delphai** [comment, marked as answer]
You need to set `autoResetSelectedRows: false` and you need to add `getRowId: (row) => row.key` so it doesn't use idx as id and of course add a field `key` to your data structure and make sure the keys are unique.

https://codesandbox.io/s/react-table-server-side-pagination-row-select-5gfrp

**Reply by ignlopezsanchez** [reply]
thanks!!!

**Reply by qmup** [reply]
Help! I need to get all selected rows with this example.
For example: I selected all rows (10 records) in the first page, then I selected all rows(10 records) in the second page.
I'll try `selectedFlatRows.map((r) => r.original)` but the result is only 10 records in the second page and I need to get all 20 rows.
Do you have any ideas?

**Reply by LeshaZholner** [reply]
> Help! I need to get all selected rows with this example. For example: I selected all rows (10 records) in the first page, then I selected all rows(10 records) in the second page. I'll try `selectedFlatRows.map((r) => r.original)` but the result is only 10 records in the second page and I need to get all 20 rows. Do you have any ideas?

You can get all selected ids from state. state.selectedRowIds :)

**Reply by timB73** [reply]
how do you do this with tanstack table v8? `autoResetSelectedRows` doesn't seem to be an option

**Reply by vanisov** [reply]
You have solved all my struggles and problems. Thank you kind sir!

**Reply by miltonlast** [reply]
@timB73 try this 
```ts
  table.setOptions((prev) => ({
    ...prev,
    getRowId: (row) => row.key,
    autoResetSelectedRows: false,
  }));
```
